### PR TITLE
arch: move common app_data_alignment.ld file

### PR DESCRIPTION
**Pull request by nashif**
This file is being used by different architectures and is not ARC
specific.

### CODE REVIEW

**Comment by andrewboie** [comment]
Hmm, don't merge this quite yet, I'm knee-deep in working on app shmem and will follow up on this later in the week

File: CMakeLists.txt [hunk 1]
```diff
@@ -1199,13 +1199,17 @@ zephyr_compile_options(
...
 configure_file(
      $ENV{ZEPHYR_BASE}/include/arch/arc/v2/app_data_alignment.ld
      ${PROJECT_BINARY_DIR}/include/generated/app_data_alignment.ld)
+endif()
 
+if(CONFIG_ARM)
 configure_file(
      $ENV{ZEPHYR_BASE}/include/arch/arm/cortex_m/scripts/app_smem.ld
      ${PROJECT_BINARY_DIR}/include/generated/app_smem.ld)
+endif()
```
Comment (SebastianBoe) [inline CMakeLists.txt]: IIRC one of these scripts was re-used by multiple architectures, or did we fix that @ioannisg ?

Comment (ioannisg) [reply CMakeLists.txt]: The first change is fine; app_data_alignment is not used in ARM.

Comment (ioannisg) [reply CMakeLists.txt]: app_smem_ld is also used by ARC , AFAIK. So we will have a problem if we do as the patch suggests, right @SebastianBoe ?

Comment (SebastianBoe) [reply CMakeLists.txt]: Yep. That seems to be correct.

```
sebo@mach:~/ncs/zephyr$ ack app_smem.ld ~/ncs/zephyr/
ack app_smem.ld ~/ncs/zephyr/
/home/sebo/ncs/zephyr/include/arch/arm/cortex_m/scripts/linker.ld
243:		#include <app_smem.ld>

/home/sebo/ncs/zephyr/include/arch/arc/v2/linker.ld
133:		#include <app_smem.ld>

/home/sebo/ncs/zephyr/CMakeLists.txt
1024:     $ENV{ZEPHYR_BASE}/include/arch/arm/cortex_m/scripts/app_smem.ld
1025:     ${PROJECT_BINARY_DIR}/include/generated/app_smem.ld)
1030:        set(APP_SMEM_LD "${PROJECT_BINARY_DIR}/include/generated/app_smem.ld")

```

Comment (SebastianBoe) [reply CMakeLists.txt]: But it might be a moot point if @andrewboie is going to rewrite this ...

Comment (nashif) [reply CMakeLists.txt]: > app_smem_ld is also used by ARC , AFAIK. So we will have a problem if we do as the patch suggests, right @SebastianBoe ?

Then we need to move those from include/<ARCH>/ and put them in some common place.

**Comment by nashif** [comment]
@andrewboie this code was not touched in your rework of app shmem. IS this code still needed?

**Comment by andrewboie** [comment]
I am not sure can you just file a bug on this and assign to me?

**Comment by andrewboie** [comment]
recheck

File: CMakeLists.txt [hunk 2]
```diff
@@ -1199,13 +1199,17 @@ zephyr_compile_options(
   $<$<COMPILE_LANGUAGE:C>:-std=${CSTD}>
 )
 
+if(CONFIG_ARC)
 configure_file(
      $ENV{ZEPHYR_BASE}/include/arch/arc/v2/app_data_alignment.ld
      ${PROJECT_BINARY_DIR}/include/generated/app_data_alignment.ld)
+endif()
 
+if(CONFIG_ARM)
```
Comment (SebastianBoe) [inline CMakeLists.txt]: app_smem_ld is also used by ARC

**Review by SebastianBoe — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 2 above)
Comment (nashif) [reply CMakeLists.txt]: then I guess `include/arch/arm/cortex_m/scripts/app_smem.ld` should be placed in some common place.

**Review by tyrantlizard77 — APPROVED** [review]
LGTM

**Comment by andrewboie** [comment]
You should update the title/description of this PR btw

**Review by SebastianBoe — APPROVED** [review]
Isn't it app_smem.ld, and not app_data_alignment.ld that is being used by multiple architectures?

```
sebo@mach:~/ncs/zephyr$ ack app_smem.ld ~/ncs/zephyr/
/home/sebo/ncs/zephyr/include/arch/arm/cortex_m/scripts/linker.ld
375:#include <app_smem.ld>

/home/sebo/ncs/zephyr/include/arch/x86/linker.ld
215:#include <app_smem.ld>

/home/sebo/ncs/zephyr/include/arch/arc/v2/linker.ld
145:#include <app_smem.ld>

/home/sebo/ncs/zephyr/CMakeLists.txt
1147:     $ENV{ZEPHYR_BASE}/include/arch/arm/cortex_m/scripts/app_smem.ld
1148:     ${PROJECT_BINARY_DIR}/include/generated/app_smem.ld)
1151:  set(APP_SMEM_LD "${PROJECT_BINARY_DIR}/include/generated/app_smem.ld")




sebo@mach:~/ncs/zephyr$ ack app_data_alignment.ld ~/ncs/zephyr/
/home/sebo/ncs/zephyr/include/arch/arc/v2/linker.ld
139:#include <app_data_alignment.ld>

/home/sebo/ncs/zephyr/CMakeLists.txt
1143:     $ENV{ZEPHYR_BASE}/include/arch/arc/v2/app_data_alignment.ld
1144:     ${PROJECT_BINARY_DIR}/include/generated/app_data_alignment.ld)

```

**Review by tyrantlizard77 — APPROVED** [review]

**Pull request merged**